src/Hpack/Config.hs

```diff
@@ -150,11 +153,14 @@ instance FromJSON (CaptureUnknownFields FlagSection) where
 
 getUnknownFields :: forall a. HasFieldNames a => Value -> Proxy a -> [FieldName]
 getUnknownFields v _ = case v of
-  Object o -> unknown
+  Object o -> ignoreUnderscored unknown
     where
       unknown = keys \\ fields
       keys = map T.unpack (HashMap.keys o)
       fields = fieldNames (Proxy :: Proxy a)
+      ignoreUnderscored
+        | warnUnderscoredUnknownFields (Proxy :: Proxy a) = id
+        | otherwise = filter (not . isPrefixOf "_")
```

Comment:
Oh, I had almost the exact same code but then I realized we agreed on top-level unknown fields and rewrote it this way. :-(
But sure, I can simplify this.